models/spring-ai-openai/src/main/java/org/springframework/ai/openai/OpenAiTranscriptionClient.java

```diff
@@ -0,0 +1,140 @@
+										* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+										* See the License for the specific language governing permissions and
+										* limitations under the License.
+										*/
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.ai.chat.ChatOptions;
+import org.springframework.ai.chat.metadata.RateLimit;
+import org.springframework.ai.model.ModelOptionsUtils;
+import org.springframework.ai.openai.api.OpenAiApi;
+import org.springframework.ai.openai.api.OpenAiApi.OpenAiApiException;
+import org.springframework.ai.openai.metadata.OpenAiTranscriptionResponseMetadata;
+import org.springframework.ai.openai.metadata.support.OpenAiResponseHeaderExtractor;
+import org.springframework.ai.transcription.*;
+import org.springframework.core.io.Resource;
+import org.springframework.http.ResponseEntity;
+import org.springframework.retry.support.RetryTemplate;
+import org.springframework.util.Assert;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+
+import java.time.Duration;
+import java.util.List;
+
+/**
+ * {@link TranscriptionClient} implementation for {@literal OpenAI} backed by
+ * {@link OpenAiApi}.
+ *
+ * @author Michael Lavelle
+ * @see TranscriptionClient
+ * @see OpenAiApi
+ */
+public class OpenAiTranscriptionClient implements TranscriptionClient {
+
+	private final Logger logger = LoggerFactory.getLogger(getClass());
+
+	private OpenAiTranscriptionOptions defaultOptions = OpenAiTranscriptionOptions.builder()
+		.withModel("whisper-1")
+		.withTemperature(0.7f)
+		.build();
+
+	public final RetryTemplate retryTemplate = RetryTemplate.builder()
+		.maxAttempts(10)
+		.retryOn(OpenAiApiException.class)
+		.exponentialBackoff(Duration.ofMillis(2000), 5, Duration.ofMillis(3 * 60000))
+		.build();
+
+	private final OpenAiApi openAiApi;
+
+	public OpenAiTranscriptionClient(OpenAiApi openAiApi) {
+		Assert.notNull(openAiApi, "OpenAiApi must not be null");
+		this.openAiApi = openAiApi;
+	}
+
+	public OpenAiTranscriptionClient withDefaultOptions(OpenAiTranscriptionOptions options) {
+		this.defaultOptions = options;
+		return this;
+	}
+
+	@Override
+	public TranscriptionResponse call(TranscriptionRequest request) {
+
+		return this.retryTemplate.execute(ctx -> {
+			Resource audioResource = request.getInstructions();
+
+			MultiValueMap<String, Object> reqyestBody = createRequestBody(request);
+
+			ResponseEntity<OpenAiApi.Transcription> transcriptionEntity = this.openAiApi
+				.transcriptionEntity(reqyestBody);
+
+			var transcription = transcriptionEntity.getBody();
+
+			if (transcription == null) {
+				logger.warn("No transcription returned for request: {}", audioResource);
+				return new TranscriptionResponse(null);
+			}
+
+			Transcript transcript = new Transcript(transcription.text());
+
+			RateLimit rateLimits = OpenAiResponseHeaderExtractor.extractAiResponseHeaders(transcriptionEntity);
```

Comment:
I'm re-using these RateLimit and Usage classes from the chat packages here, but given that these are now used outside of chat, should they be moved into a more generic package ?

Reply:
I am not yet sure, we need to review this in the 0.9.0 timeline.  I suspect there are cross cutting concerns.  Now we are wrapping up the other direction, input options, to support a tiered structure of portable and model specific options.  The same analysis and likely same approach needs to apply for the metadata info coming out of the response.